### PR TITLE
[ISSUE #4389] Optimize for MixAll#properties2Object

### PR DESCRIPTION
**Pull request by dugenkui03**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

XXXXX

## Brief change log
Optimize for `MixAll#properties2Object`:

## Brief change log

1. Add debug log for exception;
2. Add  processing logic for `byte`, `short` and `char`;
3. Reorder `if ... else`.

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by HScarb** [comment]
Is it necessary to add parsing of new types?
As many of the fields are `String`, will more `else if` branches cost more time to parse `String` fields?

**Review by caigy — CHANGES_REQUESTED** [review]
It seems `MixAll.properties2Object()` has been modified, but test cases are about `MixAll.object2Properties()`

**Comment by MatrixHB** [comment]
Related to [ISSUE #4389](https://github.com/apache/rocketmq/issues/4389)

File: common/src/test/java/org/apache/rocketmq/common/UtilAllTest.java [hunk 1]
```diff
@@ -60,11 +60,17 @@ public void testProperties2String() {
         demoConfig.setDemoWidth(456);
         demoConfig.setDemoName("TestDemo");
         demoConfig.setDemoOK(true);
+        demoConfig.setDemoByte((byte) 78);
+        demoConfig.setDemoShort((short) 90);
+        demoConfig.setDemoChar('c');
         Properties properties = MixAll.object2Properties(demoConfig);
         assertThat(properties.getProperty("demoLength")).isEqualTo("123");
         assertThat(properties.getProperty("demoWidth")).isEqualTo("456");
         assertThat(properties.getProperty("demoOK")).isEqualTo("true");
```
Comment (xiaoyifang) [inline common/src/test/java/org/apache/rocketmq/common/UtilAllTest.java:69]: you have modified properties2Object,  but test object2Properties method?

**Pull request closed without merging**